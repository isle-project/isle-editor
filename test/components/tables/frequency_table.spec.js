// MODULES //

import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import FrequencyTable from '@isle-project/components/tables/frequency-table';
import heartdisease from './../fixtures/heartdisease.json';


// TESTS //

describe( '<FrequencyTable />', function test() {
	it( 'renders a frequency table', () => {
		const table = renderer.create( <FrequencyTable
			data={heartdisease}
			variable="Drugs"
		/> );
		expect( table ).toMatchSnapshot();
	});
});
