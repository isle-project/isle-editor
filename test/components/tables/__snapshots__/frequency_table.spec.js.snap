// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FrequencyTable /> renders a frequency table 1`] = `
<table
  className="sortable-theme-bootstrap table table-sm table-bordered"
  data-sortable={true}
>
  <thead>
    <tr>
      <th
        className="not-sortable"
      >
        Drugs
      </th>
      <th>
        category
      </th>
      <th>
        count
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td />
      <td>
        0
      </td>
      <td>
        610
      </td>
    </tr>
    <tr>
      <td />
      <td>
        1
      </td>
      <td>
        89
      </td>
    </tr>
    <tr>
      <td />
      <td>
        2
      </td>
      <td>
        89
      </td>
    </tr>
  </tbody>
  <tbody>
    <tr>
      <th>
        total
      </th>
      <td />
      <td>
        788
      </td>
    </tr>
  </tbody>
</table>
`;
